i6_models/parts/conformer/mhsa.py

```diff
@@ -39,17 +39,20 @@ def __init__(self, cfg: ConformerMHSAV1Config):
         )
         self.dropout = cfg.dropout
 
-    def forward(self, input_tensor: torch.Tensor, key_padding_mask: Optional[torch.Tensor] = None) -> torch.Tensor:
+    def forward(self, input_tensor: torch.Tensor, sequence_mask: Optional[torch.Tensor] = None) -> torch.Tensor:
         """
         Apply layer norm and multi-head self attention and dropout
-        :param Optional[torch.Tensor] key_padding_mask: could be a binary or float mask of shape (B, T)
+        :param Optional[torch.Tensor] sequence_mask: could be a binary or float mask of shape (B, T), 1 signals within sequence, 0 outside, will be inverted to match the torch.nn.MultiheadAttention module
         which will be applied/added to dot product, used to mask padded key positions out
         """
-
+        if sequence_mask is not None:
+            inv_sequence_mask = (1 - sequence_mask.float()).bool()
```

Comment:
is it concerning that you subtract float mask from integer 1?

how about 
```suggestion
            inv_sequence_mask = torch.logical_not(sequence_mask)
```

Reply:
I thought about that too, but there was no clear agreement (yet?) whether the mask is always bool, that I am aware of. So here it might also be float (c.f. #17). But if you prefer logical_not I could always cast the mask to bool beforehand.

Reply:
according to the examples at https://pytorch.org/docs/stable/generated/torch.logical_not.html `torch.logical_not` also handles float and integer inputs (treating 0 and 0.0 as False and all other as True)

Reply:
Oh nice, I did not know that.